subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationWorker.java

```diff
@@ -16,16 +16,18 @@
 
 package org.gradle.internal.operations;
 
-import org.gradle.api.Action;
+import org.gradle.api.Nullable;
 
 /**
- * A worker that can run build operations of type {@link T}.
+ * A worker that can run build operations.
  *
  * Implementations must be thread-safe.
  */
-public interface BuildOperationWorker<T extends BuildOperation> extends Action<T> {
+public interface BuildOperationWorker<O extends BuildOperation> {
```

Comment:
What about using `Tool` then? We already have a `Tool` interface in native, which is describing how to call an external tools (args etc.).

Maybe `ExecutableTool`?